espresso/submitter/three_polling_run_loops.go

```diff
@@ -0,0 +1,833 @@
+	espressoLastConfirmedPos     []byte = []byte("_espressoLastConfirmedPos") // contains the position of the last confirmed message
+)
+
+// ThreePollingRunLoops is a struct that implements the EspressoSubmitter interface
+// and provides methods to submit transactions to Espresso, manage pending.
+type ThreePollingRunLoops struct {
+	espressoPendingTxnPosMutex sync.Mutex
+	espressoSubmittedTxnsMutex sync.Mutex
+
+	db                 ethdb.Database
+	messageGetter      MessageGetter
+	espressoClient     espresso_client.EspressoClient
+	lightClientReader  espresso_light_client.LightClientReaderInterface
+	espressoKeyManager espresso_key_manager.EspressoKeyManagerInterface
+
+	userDataAttestationFile string
+	quoteFile               string
+
+	chainID                               uint64
+	espressoTxnsPollingInterval           time.Duration
+	espressoTxnsSendingInterval           time.Duration
+	espressoTxnsResubmissionInterval      time.Duration
+	maxBlockLagBeforeEscapeHatch          uint64
+	espressoMaxTransactionSize            int64
+	resubmitEspressoTxDeadline            time.Duration
+	lastSubmitFailureAt                   *time.Time
+	UseEscapeHatch                        bool
+	EscapeHatchEnabled                    bool
+	InitialFinalizedSequencerMessageCount *big.Int
+}
+
+var _ EspressoSubmitter = &ThreePollingRunLoops{}
+
+func NewThreePollingRunLoops(options ...EspressoSubmitterConfigOption) (EspressoSubmitter, error) {
+	config := DefaultEspressoSubmitterConfig
+
+	for _, option := range options {
+		option(&config)
+	}
+
+	// Validate the config, to ensure we have everything we need to move
+	// forward
+	if err := ValidateEspressoSubmitterConfig(config); err != nil {
+		return nil, fmt.Errorf("invalid espresso submitter config: %w", err)
+	}
+
+	return &ThreePollingRunLoops{
+		db:                 config.Db,
+		messageGetter:      config.MessageGetter,
+		espressoClient:     config.EspressoClient,
+		lightClientReader:  config.LightClientReader,
+		espressoKeyManager: config.KeyManager,
+
+		userDataAttestationFile: config.UserDataAttestationFile,
+		quoteFile:               config.QuoteFile,
+
+		espressoTxnsPollingInterval:      config.EspressoTxnsPollingInterval,
+		espressoTxnsSendingInterval:      config.EspressoTxnSendingInterval,
+		espressoTxnsResubmissionInterval: config.EspressoTxnsResubmissionInterval,
+		maxBlockLagBeforeEscapeHatch:     config.MaxBlockLagBeforeEscapeHatch,
+		espressoMaxTransactionSize:       config.EspressoMaxTransactionSize,
+		resubmitEspressoTxDeadline:       config.ResubmitEspressoTxDeadline,
+		EscapeHatchEnabled:               config.EscapeHatchEnabled,
+		UseEscapeHatch:                   config.UseEscapeHatch,
+
+		InitialFinalizedSequencerMessageCount: config.InitialFinalizedSequencerMessageCount,
+	}, nil
+}
+
+func (s *ThreePollingRunLoops) enqueuePendingTransaction(pos []arbutil.MessageIndex) error {
```

Comment:
I wonder if we even need this function, its only being called in one place. I think we can just remove this

Reply:
I agree.

However, I am attempting to keep our existing code as-is in order to reflect a code relocation, rather than any major behavior changes.